account_statement_base/views/account_bank_statement_line.xml

```diff
@@ -0,0 +1,158 @@
+                        <field name="payment_ids" widget="many2many_tags" />
+                    </group>
+                </page>
+            </notebook>
+            <footer invisible="not context.get('add_bank_statement_line')">
+                <button name="save" type="object" string="Save" class="btn-primary" />
+                <button string="Cancel" special="cancel" />
+            </footer>
+        </form>
+    </field>
+</record>
+
+<record id="account_bank_statement_line_tree" model="ir.ui.view">
+    <field name="model">account.bank.statement.line</field>
+    <field name="arch" type="xml">
+        <tree decoration-muted="is_reconciled">
+            <field name="sequence" widget="handle" />
+            <field name="date" />
+            <field name="payment_ref" />
+            <field name="ref" optional="hide" />
+            <field name="partner_id" />
+            <field name="amount" />
+            <field
+                    name="amount_currency"
+                    optional="hide"
+                    groups="base.group_multi_currency"
+                />
+            <field name="narration" optional="hide" string="Notes" />
+            <field name="journal_id" optional="hide" />
+            <field name="partner_bank_id" optional="hide" />
+            <button
+                    name="action_undo_reconciliation"
+                    type="object"
+                    string="Revert reconciliation"
+                    icon="fa-undo"
+                    attrs="{'invisible': [('is_reconciled', '=', False)]}"
+                />
+            <field name="company_id" invisible="1" />
+            <field name="is_reconciled" invisible="1" />
+            <field name="currency_id" invisible="1" />
+            <field name="foreign_currency_id" invisible="1" />
+            <field name="country_code" invisible="1" />
+            <field name="state" invisible="1" />
+        </tree>
+    </field>
+</record>
+
+<record id="account_bank_statement_line_search" model="ir.ui.view">
+    <field name="model">account.bank.statement.line</field>
+    <field name="arch" type="xml">
+        <search>
+            <field
+                    name="payment_ref"
+                    filter_domain="['|', '|', ('payment_ref', 'ilike', self), ('ref', 'ilike', self), ('narration', 'ilike', self)]"
+                    string="Label, Ref or Notes"
+                />
+            <field name="partner_id" />
+            <field name="date" />
+            <field name="journal_id" domain="[('type', 'in', ('bank', 'cash'))]" />
+            <separator />
+            <filter
+                    name="reconciled"
+                    string="Reconciled"
+                    domain="[('is_reconciled', '=', True)]"
+                />
+            <filter
+                    name="not_reconciled"
+                    string="Not Reconciled"
+                    domain="[('is_reconciled', '=', False)]"
+                />
```

Comment:
I think the v15 search view has a few more handy filters: what about adding narration and amount https://github.com/odoo/odoo/blob/15.0/addons/account/views/account_bank_statement_views.xml#L279

Reply:
I added search on "amount". Search on "narration" is already included in the first filter, see "filter_domain"